CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#-----------------------------------------------------------------------
+# CMake preamble.
+
+# FindPython's Development.Module component was added in 3.18, and
+# pybind11 recommends 3.18.2+.
+cmake_minimum_required(VERSION 3.18.2)
+
+# Additional include directories for CMake utils.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+if (APPLE AND NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
+    # The value of this variable should be set prior to the first
+    # project() command invocation because it may influence
+    # configuration of the toolchain and flags.
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13" CACHE STRING "Minimum OS X deployment version")
```

Comment:
There was something that @foundrytom mentioned in a meeting earlier we were both in, that OAIO was going to target VFXCY22 initially.
https://vfxplatform.com/ on the CY22 column says minimum deployment is 10.15.

Reply:
Checked with @foundrytom - we may yet support CY21, which is 10.13 - but still C++17 :tada: !

Reply:
[fix] Given we're not building `macOS` yet - and have the other `APPLE` stuff as TODOs can we just omit this for now?

Reply:
OK, replaced with a TODO in 8b540fee